### PR TITLE
Survey delete: fixed app shows login page

### LINKED ISSUE #658: On survey delete, app shows login
After deleting a survey, app shows login page

tested on arena test.

### PR DESCRIPTION
**Pull request by SteRiccio**
Fixes #658

### CODE REVIEW

File: webapp/loggedin/modules/home/dashboard/surveyInfo/surveyInfo.js [hunk 1]
```diff
@@ -19,13 +19,17 @@ import { deleteSurvey, publishSurvey } from '@webapp/survey/actions'
...
-
+const SurveyInfo = () => {
   const [showDeleteDialog, setShowDeleteDialog] = useState(false)
 
   const i18n = useI18n()
   const dispatch = useDispatch()
+  const history = useHistory()
+
+  const user = useSelector(AppState.getUser)
+  const surveyInfo = useSelector(SurveyState.getSurveyInfo)
```
Comment (minotogna) [inline webapp/loggedin/modules/home/dashboard/surveyInfo/surveyInfo.js:30]: should we add two hooks ```useUser``` and ```useSurveyInfo```?

File: webapp/loggedin/modules/home/dashboard/surveyInfo/surveyInfo.js [hunk 2]
```diff
@@ -19,13 +19,17 @@ import { deleteSurvey, publishSurvey } from '@webapp/survey/actions'
...
+  const canEditDef = Authorizer.canEditSurvey(user, surveyInfo)
```
Comment (minotogna) [inline webapp/loggedin/modules/home/dashboard/surveyInfo/surveyInfo.js:31]: and what about a ```useAuthCanEditSurvey```? defined in useAuth hook?

File: webapp/loggedin/modules/home/dashboard/surveyInfo/surveyInfo.js (hunk 1 above)
Comment (SteRiccio) [reply webapp/loggedin/modules/home/dashboard/surveyInfo/surveyInfo.js:30]: ok

File: webapp/loggedin/modules/home/dashboard/surveyInfo/surveyInfo.js (hunk 2 above)
Comment (SteRiccio) [reply webapp/loggedin/modules/home/dashboard/surveyInfo/surveyInfo.js:31]: ok

File: webapp/loggedin/modules/designer/designerView.js [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
 import CategoriesView from '@webapp/loggedin/surveyViews/categories/categoriesView'
 import CategoryView from '@webapp/loggedin/surveyViews/category/categoryView'
 import TaxonomiesView from '@webapp/loggedin/surveyViews/taxonomies/taxonomiesView'
 import TaxonomyView from '@webapp/loggedin/surveyViews/taxonomy/taxonomyView'
-import SurveyHierarchyView from './surveyHierarchy/surveyHierarchyView'
 
-import * as AppState from '@webapp/app/appState'
-import * as SurveyState from '@webapp/survey/surveyState'
 import { appModules, appModuleUri, designerModules } from '@webapp/app/appModules'
+import { useAuthCanEditSurvey } from '@webapp/commonComponents/hooks/useAuth'
```
Comment (minotogna) [inline webapp/loggedin/modules/designer/designerView.js:13]: export this from hooks/index.js

Comment (SteRiccio) [reply webapp/loggedin/modules/designer/designerView.js:13]: ok

**Pull request merged**